src/dumbo/src/tcp/endpoint.rs

```diff
@@ -299,16 +303,36 @@ impl Endpoint {
     }
 }
 
+/// `parse_request` parses the request bytes and builds a `micro_http::Response` by the given callback function.
+pub fn parse_request(
+    request_bytes: &[u8],
+    callback: fn(Result<Request, RequestError>) -> Response,
+) -> Response {
+    let request = Request::try_from(request_bytes);
+    callback(request)
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
 
     use std::fmt;
     use std::str::from_utf8;
 
+    use micro_http::{Request, RequestError, Response, StatusCode, Version};
     use pdu::tcp::Flags as TcpFlags;
     use tcp::connection::tests::ConnectionTester;
 
+    fn mock_callback(request: Result<Request, RequestError>) -> Response {
```

Comment:
Even the mock callback should process an already obtained `micro_http::Request` and return a correspondent response.
```suggestion
    fn mock_callback(request: Request) -> Response {
```